queries.sql

```diff
@@ -5,4 +5,67 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = TRUE;
 SELECT * FROM animals WHERE name <> 'Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL OR species = '';
+
+BEGIN;
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+-- Start a transaction
+BEGIN;
+
+-- Delete all animals born after Jan 1st, 2022
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+-- Create a savepoint for the transaction
+SAVEPOINT weight_update_savepoint;
+
+-- Update all animals' weight to be their weight multiplied by -1
+UPDATE animals SET weight_kg = -weight_kg;
+
+-- Rollback to the savepoint
+ROLLBACK TO SAVEPOINT weight_update_savepoint;
+
+-- Update all animals' weights that are negative to be their weight multiplied by -1
+UPDATE animals SET weight_kg = -weight_kg WHERE weight_kg < 0;
+
+-- Commit the transaction
+COMMIT;
+
+SELECT COUNT(*) AS animal_count FROM animals;
+
+SELECT COUNT(*) AS animals_never_tried_to_escape_count
+FROM animals
+WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) AS avg_weight FROM animals;
+
+SELECT neutered, COUNT(*) AS escape_count
+FROM animals
+WHERE escape_attempts > 0
+GROUP BY neutered
+ORDER BY escape_count DESC;
+
+SELECT
+    species,
+    MIN(weight_kg) AS min_weight,
+    MAX(weight_kg) AS max_weight
+FROM
+    animals
+GROUP BY
+    species;
+
+SELECT
+    species,
+    AVG(escape_attempts) AS avg_escape_attempts
+FROM
+    animals
+WHERE
+    date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY
+    species;
```

Comment:
- [ ] We are missing a query from this file which sets a animal's specie as null

```sql
UPDATE animals
   SET species = 'unspecified'; 
```